File: go/lc/lcp61.go
package main

func temperatureTrend(temperatureA []int, temperatureB []int) int {
	cmp := func(a, b int) int {
		if a > b {
			return 1
		} else if a < b {
			return -1
		} else {
			return 0
		}
	}

	max := func(a, b int) int {
		if a >= b {
			return a
		} else {
			return b
		}
	}

	res, tmp := 0, 0
	for i := 1; i < len(temperatureA); i++ {
		if cmp(temperatureA[i], temperatureA[i-1])-cmp(temperatureB[i], temperatureB[i-1]) == 0 {
			tmp++
		} else {
			res = max(res, tmp)
			tmp = 0
		}
	}

	return max(res, tmp)
}

File: java/com/lc/lcp61.java
package lc;

import java.util.*;

public class lcp61 {
    public int temperatureTrend(int[] temperatureA, int[] temperatureB) {
        int res = 0, tmp = 0;
        for (int i = 1; i < temperatureA.length; i++) {
            if (cmp(temperatureA[i - 1], temperatureA[i]) - cmp(temperatureB[i - 1], temperatureB[i]) == 0) {
                tmp++;
            } else {
                res = Math.max(tmp, res);
                tmp = 0;
            }
        }
        return Math.max(tmp, res);
    }

    private int cmp(int a, int b) {
        if (a == b) {
            return 0;
        } else if (a < b) {
            return 1;
        } else {
            return -1;
        }
    }
}
